lib/swaps/Swaps.ts

```diff
@@ -274,17 +287,30 @@ class Swaps extends EventEmitter {
     };
     this.sanitySwaps.set(rHash, sanitySwap);
 
-    await peer.sendPacket(new packets.SanitySwapPacket({
+    const sanitySwapPacket = new packets.SanitySwapPacket({
       currency,
       rHash,
-    }));
+    });
+
+    try {
+      await Promise.all([
+        swapClient.addInvoice(rHash, 1),
+        peer.sendPacket(sanitySwapPacket),
```

Comment:
`AddInvoice` and `SendPacket` should happen in 2 separated steps, to avoid race, no?

Reply:
I didn't think so, since the sanity swap won't actually begin until we receive the sanity swap ack packet. But thinking about it some more, it's indeed a race that could go wrong if the peer adds its invoice and respond to us faster than our add invoice call. So I'll put in some logic to ensure we don't begin the sanity swap until our invoice has been added, but I think we can still start both these calls in parallel (or if that's too complicated, just await these here separately).

Reply:
Actually looking at the code again, I think this is fine. We are awaiting the `addInvoice` call before we begin the swap with `sendSmallestAmount` a few lines down, so there's no risk that we begin the swap before lnd knows about our invoice.